### PR TITLE
fix(judicial-system): fix prefill for ruling input

### PR DESCRIPTION
**Pull request by Arnarkari93**
# ...

[Asana task](https://app.asana.com/0/1199153462262248/1201918336079019/f)

## What

- Prefill input when the judge is drafting the ruling 

## Why

- It is prefilled later in the flow with the name of the judge

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/judicial-system/web/src/components/RulingInput/RulingInput.tsx [hunk 1]
```diff
@@ -14,41 +14,32 @@ import {
...
-    isRequired,
-    rows,
-  } = props
+  const { workingCase, setWorkingCase, isRequired, rows } = props
   const { updateCase, autofill } = useCase()
   const { formatMessage } = useIntl()
   const [rulingErrorMessage, setRulingErrorMessage] = useState('')
 
   useEffect(() => {
-    if (isCaseUpToDate) {
```
Comment (Arnarkari93) [inline apps/judicial-system/web/src/components/RulingInput/RulingInput.tsx:35]: isCaseUpToDate is set to false after one sec. Therefore, it is never set to `true` in modals.

**Review by oddsson — CHANGES_REQUESTED** [review]
I agree with you, `isCaseUpToDate` will never be true in modals. However the `rulingInput` component is not only used in this modal, it's used on the ruling pages as well, and on those pages we need the `isCaseUpToDate` functionality. 

Maybe we should move the autofill out of the rulingInput component and onto the overview page? That way, the `isCaseUpToDate` functionality continues to work. What do you think?

**Comment by Arnarkari93** [comment]
> I agree with you, `isCaseUpToDate` will never be true in modals. However the `rulingInput` component is not only used in this modal, it's used on the ruling pages as well, and on those pages we need the `isCaseUpToDate` functionality.
>   
> Maybe we should move the autofill out of the rulingInput component and onto the overview page? That way, the `isCaseUpToDate` functionality continues to work. What do you think?

I was thinking if we should rethink the autofill logic. Feels a bit hacky to control with a state that is reset with a `setTimeout`. I was thinking we could write a custom hook that we use for each field that needs to be autofill and make sure there it is only updated when we want it to.

**Comment by Arnarkari93** [comment]
Close in favour of #6817

**Pull request closed without merging**